systemtests/src/main/java/com/github/streamshub/systemtests/utils/resourceutils/KafkaTopicUtils.java

```diff
@@ -0,0 +1,201 @@
+package com.github.streamshub.systemtests.utils.resourceutils;
+
+import com.github.streamshub.systemtests.clients.KafkaClients;
+import com.github.streamshub.systemtests.constants.Constants;
+import com.github.streamshub.systemtests.constants.Labels;
+import com.github.streamshub.systemtests.logs.LogWrapper;
+import com.github.streamshub.systemtests.utils.WaitUtils;
+import io.fabric8.kubernetes.api.model.Pod;
+import io.skodjob.testframe.resources.KubeResourceManager;
+import io.strimzi.api.ResourceAnnotations;
+import io.strimzi.api.ResourceLabels;
+import io.strimzi.api.kafka.model.nodepool.KafkaNodePool;
+import io.strimzi.api.kafka.model.nodepool.KafkaNodePoolBuilder;
+import io.strimzi.api.kafka.model.topic.KafkaTopic;
+import io.strimzi.api.kafka.model.topic.KafkaTopicBuilder;
+import org.apache.logging.log4j.Logger;
+
+import java.util.List;
+import java.util.stream.IntStream;
+
+public class KafkaTopicUtils {
+    private static final Logger LOGGER = LogWrapper.getLogger(KafkaTopicUtils.class);
+    
+    private KafkaTopicUtils() {}
+    
+    public static List<KafkaTopic> createTopicsAndReturn(String namespace, String kafkaName, String topicNamePrefix, int numberToCreate, boolean waitForTopics, int partitions, int replicas, int minIsr) {
+        LOGGER.info("Create {} topics for cluster {} with topic name prefix {}", numberToCreate, kafkaName, topicNamePrefix);
+
+        List<KafkaTopic> topics = IntStream.range(0, numberToCreate)
+            .mapToObj(i -> defaultTopic(namespace, kafkaName, topicNamePrefix + "-" + i, partitions, replicas, minIsr).build())
+            .toList();
+
+        if (waitForTopics) {
+            topics.forEach(KubeResourceManager.get()::createResourceWithWait);
```

Comment:
instead of for each, which will do this sequently, you can just pass the list to array and use the async method probably?

Reply:
yea, sure, i do it on other places so i can unify this